server/storage/database_test.go

```diff
@@ -2,290 +2,326 @@ package storage
 	"testing"
 
-	"github.com/DATA-DOG/go-sqlmock"
-	"github.com/go-sql-driver/mysql"
+	"github.com/endophage/gotuf/data"
+	"github.com/jinzhu/gorm"
+	_ "github.com/mattn/go-sqlite3"
 	"github.com/stretchr/testify/assert"
 )
 
-func TestMySQLUpdateCurrent(t *testing.T) {
-	db, err := sqlmock.New()
-	assert.Nil(t, err, "Could not initialize mock DB")
-	s := NewMySQLStorage(db)
-	update := MetaUpdate{
-		Role:    "root",
-		Version: 0,
-		Data:    []byte("1"),
-	}
-	sqlmock.ExpectQuery("SELECT count\\(\\*\\) FROM `tuf_files` WHERE `gun`=\\? AND `role`=\\? AND `version`>=\\?\\;").WithArgs(
-		"testGUN",
-		update.Role,
-		update.Version,
-	).WillReturnRows(sqlmock.RowsFromCSVString([]string{"count(*)"}, "0"))
-	sqlmock.ExpectExec("INSERT INTO `tuf_files` \\(`gun`, `role`, `version`, `data`\\) VALUES \\(\\?,\\?,\\?,\\?\\);").WithArgs(
-		"testGUN",
-		update.Role,
-		update.Version,
-		update.Data,
-	).WillReturnResult(sqlmock.NewResult(0, 1))
-
-	err = s.UpdateCurrent(
-		"testGUN",
-		update,
-	)
-	assert.Nil(t, err, "UpdateCurrent errored unexpectedly: %v", err)
-
-	// There's a bug in the mock lib
-	//err = db.Close()
-	//assert.Nil(t, err, "Expectation not met: %v", err)
+// GormTUFFile represents a TUF file in the database
+type GormTUFFile struct {
+	ID      int    `sql:"AUTO_INCREMENT" gorm:"primary_key"`
+	Gun     string `sql:"type:varchar(255);not null"`
+	Role    string `sql:"type:varchar(255);not null"`
+	Version int
+	Data    []byte `sql:"type:longblob"`
 }
 
-func TestMySQLUpdateCurrentError(t *testing.T) {
-	db, err := sqlmock.New()
-	assert.Nil(t, err, "Could not initialize mock DB")
-	s := NewMySQLStorage(db)
-	update := MetaUpdate{
-		Role:    "root",
-		Version: 0,
-		Data:    []byte("1"),
-	}
-	sqlmock.ExpectQuery("SELECT count\\(\\*\\) FROM `tuf_files` WHERE `gun`=\\? AND `role`=\\? AND `version`>=\\?\\;").WithArgs(
-		"testGUN",
-		update.Role,
-		update.Version,
-	).WillReturnRows(sqlmock.RowsFromCSVString([]string{"count(*)"}, "0"))
-	sqlmock.ExpectExec("INSERT INTO `tuf_files` \\(`gun`, `role`, `version`, `data`\\) VALUES \\(\\?,\\?,\\?,\\?\\);").WithArgs(
-		"testGUN",
-		update.Role,
-		update.Version,
-		update.Data,
-	).WillReturnError(
-		&mysql.MySQLError{
-			Number:  1022,
-			Message: "Duplicate key error",
-		},
-	)
-
-	err = s.UpdateCurrent(
-		"testGUN",
-		update,
-	)
-	assert.NotNil(t, err, "Error should not be nil")
-	assert.IsType(t, &ErrOldVersion{}, err, "Expected ErrOldVersion error type, got: %v", err)
-
-	// There's a bug in the mock lib
-	//err = db.Close()
-	//assert.Nil(t, err, "Expectation not met: %v", err)
+// TableName sets a specific table name for GormTUFFile
+func (g GormTUFFile) TableName() string {
+	return "tuf_files"
 }
 
-func TestMySQLUpdateMany(t *testing.T) {
-	db, err := sqlmock.New()
-	assert.Nil(t, err, "Could not initialize mock DB")
-	s := NewMySQLStorage(db)
-	update1 := MetaUpdate{
+// GormTimestampKey represents a single timestamp key in the database
+type GormTimestampKey struct {
```

Comment:
There is currently no unique constraint on the `timestamp_keys` structure - I think uniqueness was enforced by the GUN being the primary key.   To clarify, do you mean add a unique constraint to the Gun column?

Also, would this mean that this is incompatible with any existing tables?  (can you migrate from one primary key to another, and if so, should we be writing those migrations?)